openbb_sdk/sdk/provider/openbb_provider/abstract/query_params.py

```diff
@@ -1,18 +1,18 @@
 """The OpenBB Standardized QueryParams Model that holds the query input parameters."""
-from pydantic import BaseModel
+from typing import Dict
 
-from openbb_provider.utils.helpers import to_snake_case
+from pydantic import BaseModel, ConfigDict, alias_generators
 
 
 class QueryParams(BaseModel):
     """The OpenBB Standardized QueryParams Model that holds the query input parameters."""
 
-    class Config:
-        """Pydantic Config."""
-
-        allow_population_by_field_name = True
-        alias_generator = to_snake_case
+    __alias_dict__: Dict[str, str] = {}
 
     def __repr__(self):
         """Return the string representation of the QueryParams object."""
-        return f"{self.__class__.__name__}({', '.join([f'{k}={v}' for k, v in self.dict().items()])})"
+        return f"{self.__class__.__name__}({', '.join([f'{k}={v}' for k, v in self.model_dump().items()])})"
+
+    model_config = ConfigDict(
+        extra="allow", populate_by_name=True, alias_generator=alias_generators.to_snake
+    )
```

Comment:
```suggestion
    )
    
        def model_dump(self, *args, **kwargs):
        original = super().model_dump(*args, **kwargs)
        by_alias = kwargs.get("by_alias", False)
        if by_alias:
            return {
                self.__alias_dict__.get(key, key): value
                for key, value in original.items()
            }
        return original
```

If we add this one here we can `model_dump(by_alias=True)` when exporting the query to send to the provider's API, wdyt?

Reply:
nice this is a good implementation 

```python
    model_config = ConfigDict(
        extra="allow", populate_by_name=True, alias_generator=alias_generators.to_snake
    )

    def model_dump(self, *args, **kwargs):
        by_alias = kwargs.pop("by_alias", False)
        original = super().model_dump(*args, **kwargs)
        if by_alias:
            return {
                self.__alias_dict__.get(key, key): value
                for key, value in original.items()
            }
        return original
```

just tested and allows `news?provider=benzinga&display_output=full&channels=All&page_size=50&symbols=PLTR` to work using snakecase

Reply:
I think we can set different alias_generator based on the providers e.g. FMP uses lower camel case and Intrinio uses snake case for their returned fields.